DiffMatchPatch.zig

```diff
@@ -255,6 +344,16 @@ const HalfMatchResult = struct {
     prefix_after: []const u8,
     suffix_after: []const u8,
     common_middle: []const u8,
+
+    // TODO maybe check for empty slice here for fewer copies,
+    // as in, maybe we can transfer ownership and replace with "".
```

Comment:
Could you explain what this comment is about? fewer copies of the slices or the memory that the slice points to?

Reply:
Sure, everything in a `half_match` struct is fresh text that we just copied, but the fate of that text is somewhat complex. 

You pointed out that a null slice is safe to free as often as you want, so part of that comment is misleading, the `deinit` code wouldn't have to check for `""`. 

I'll see about taking a pass at replacing lines which get copied over with a null slice, instead of copying them a second time.

Reply:
Yep, works fine, once I realized the step was now named coverage (which is a good change).